### PR TITLE
Add a readelf command

### PR DESCRIPTION
**Pull request by rminnich**
Sometimes, on non-Unix systems such as OSX :-), one needs readelf.

There is no convenient way to get it, and it is trivial to write in Go.

### CODE REVIEW

**Comment by rminnich** [comment]
I'm constantly amazed at how much easier it is just to write these tidbits than to figure out how to get the C ones to actually install or compile.

File: cmds/exp/readelf/readelf_test.go [hunk 1]
```diff
@@ -0,0 +1,1956 @@
...
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// Copyright 2023 the u-root Authors. All rights reserved
```
Comment (hugelgupf) [inline cmds/exp/readelf/readelf_test.go:15]: remove second copyright?

File: cmds/exp/readelf/readelf.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+func run(in io.Reader, out io.Writer, args ...string) error {
+	var (
+		files []*elf.File
+		err   error
+	)
+	switch len(args) {
+	case 0:
+		files = make([]*elf.File, 1, 1)
+		files[0], err = elf.NewFile(os.Stdin)
```
Comment (binjip978) [inline cmds/exp/readelf/readelf.go:42]: probably you want elf.NewFile(in)

File: cmds/exp/readelf/readelf_test.go [hunk 3]
```diff
@@ -0,0 +1,1956 @@
...
+// license that can be found in the LICENSE file.
+
+// .code32
+// .globl main
+// main: hlt
+// .data
+// bad: .long 0
+// gcc a.S
+
+var elfFile = []byte{
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/exp/readelf/readelf_test.go:26]: I did, but this was small enough that I did not bother. But, if you think I should, will do!

File: cmds/exp/readelf/readelf_test.go (hunk 1 above)
Comment (rminnich) [reply cmds/exp/readelf/readelf_test.go:15]: done, thanks.

File: cmds/exp/readelf/readelf_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/exp/readelf/readelf_test.go]: done, please take a look.

**Review by binjip978 — APPROVED** [review]

**Pull request merged**